pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,86 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// stages contains details about each build stage.
+	Stages []StageInfo `json:"stages,omitempty" protobuf:"bytes,11,opt,name=stages"`
+}
+
+// StageInfo contains details about a build stage.
+type StageInfo struct {
+	// Name is the identifier for each build stage.
+	Name StageName `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+
+	// StartTime identifies when this stage started.
+	StartTime unversioned.Time `json:"startTime,omitempty" protobuf:"bytes,2,opt,name=startTime"`
+
+	// DurationMilliseconds identifies how long this stage ran.
+	DurationMilliseconds int64 `json:"durationMilliseconds,omitempty" protobuf:"varint,3,opt,name=durationMilliseconds"`
+
+	// Steps contains details about each build step within a build stage.
+	Steps []StepInfo `json:"steps,omitempty" protobuf:"bytes,4,opt,name=steps"`
+}
+
+// StageName is the identifier for each build stage.
+type StageName string
+
+// Valid StageNames
+const (
+	// StageFetch fetches the source.
+	StageFetchInputs StageName = "FetchInputs"
+
+	// StagePullImages pulls the docker images.
+	StagePullImages StageName = "PullImages"
+
+	// StageBuild builds the source.
+	StageBuild StageName = "Build"
+
+	// StagePost executes post steps.
+	StagePostCommit StageName = "PostCommit"
+
+	// StagePushImage pushes the docker image.
+	StagePushImage StageName = "PushImage"
+)
+
+// StepInfo contains details about a build step.
+type StepInfo struct {
```

Comment:
API structure looks ok.  Please add more godoc to each field that describes any caveats / limitations of these fields.  Also, use the `camelCase` style (external field name) for godoc that appears on API fields since that will be part of the external API doc which is always talking about the JSON form.